d2l-search-facets-grouping.html

```diff
@@ -0,0 +1,135 @@
+<link rel="import" href="../polymer/polymer-element.html">
+<link rel="import" href="../d2l-inputs/d2l-input-checkbox-spacer.html">
+<link rel="import" href="../d2l-typography/d2l-typography-shared-styles.html">
+
+<dom-module id="d2l-search-facets-grouping">
+	<template strip-whitespace>
+		<style>
+			:host {
+				display: flex;
+				flex-direction: column;
+			}
+			.d2l-search-facets-grouping-label {
+				@apply --d2l-body-compact-text;
+				font-weight: bold;
+			}
+
+			.d2l-search-facets-grouping-show-more > button {
+				@apply --d2l-body-compact-text;
+				border: none;
+				color: var(--d2l-color-celestine);
+				font-family: inherit;
+				outline: none;
+				padding: 0px;
+			}
+
+			.d2l-search-facets-grouping-show-more > button:hover,
+			.d2l-search-facets-grouping-show-more > button:focus {
+				cursor: pointer;
+				font-weight: bold;
+			}
+
+		</style>
+		<span class="d2l-search-facets-grouping-label">[[label]]</span>
+		<d2l-input-checkbox-spacer></d2l-input-checkbox-spacer>
+		<slot></slot>
+		<template is="dom-if" if="{{_showMoreButtonVisible}}">
+			<d2l-input-checkbox-spacer class="d2l-search-facets-grouping-show-more">
+				<button on-click="_showMore">More</button>
+			</d2l-input-checkbox-spacer>
+		</template>
+	</template>
+	<script>
+		/**
+		 * `<d2l-search-facets-grouping>`
+		 * A grouping of search facet options that keeps tracks of selected options
+		 */
+		class SearchFacetsGrouping extends Polymer.Element {
+			static get is() { return 'd2l-search-facets-grouping'; }
+			static get properties() {
+				return {
+					/**
+					* List of <d2l-search-facets-option> elements
+					*/
+					_optionElements: {
+						type: Array,
+						value: function() { return []; },
+					},
+					/**
+					* List of selected options (e.g., ['optionA', ...]s)
+					*/
+					_optionsSelected: {
+						type: Array,
+						value: function() { return []; },
+					},
+					/**
+					* Visibility of 'More' button - true if initiallyShown < _optionElements.length
+					*/
+					_showMoreButtonVisible: {
+						type: Boolean,
+						value: false,
+					},
+					/**
+					* Label of the grouping
+					*/
+					label: {
+						type: String,
+						value: '',
+					},
+					/**
+					* Number of facet options initially displayed. All options are shown by default
+					*/
+					initiallyShown: {
+						type: Number,
+					}
+				};
+			}
+
+			connectedCallback() {
+				super.connectedCallback();
+				this.addEventListener('d2l-search-facets-option-change', function(e) {
```

Comment:
It looks like this event handler is added but never removed.  If this element were attached, detached, and re-attached to the DOM, 2 event handlers would be wired up so this logic would be called twice.  Best practice when wiring up event-handlers in `connectedCallback` is to also un-wire them in `disconnectedCallback`.

It may be possible to defer this work to `afterNextRender`.  I would typically do this unless there is some reason not to.